src/evict.c

```diff
@@ -412,6 +412,20 @@ int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *lev
     return C_ERR;
 }
 
+/* Return 1 if used memory is more than maxmemory after allocating more memory,
+ * return 0 if not. Redis may reject user's requests or evict some keys if used
+ * memory exceeds maxmemory, especially, when we allocate huge memory at once. */
+int overMaxmemoryAfterAlloc(size_t moremem) {
```

Comment:
It seems odd to put this in `evict.c` (as it has nothing to do with eviction).  However I realize that a lot of `evict.c` is actually about memory management.  This might be the most expedient location.

Reply:
right. we have a few other methods that interact with `maxmemory` here which are not strictly eviction. i think it's ok.